File: examples/ciflypy-examples/tests/test_nearest_dsep.py
from ciflypy_examples.nearest_dsep import find_nearest_separator

# figure from van der Zander (2019) dissertation
def test_figure_6dot2_1():
    # labels = ["Y", "V1", "V2", "V3", "Z"]
    admg = {"-->": [(0, 2), (1, 0), (1, 2), (2, 3), (3, 4)]} 
    x = 0
    y = 4

    nearest_dsep = find_nearest_separator(admg, [x], [y], [], list(range(5)))
    assert nearest_dsep is not None
    assert set(nearest_dsep) == {1, 2} or set(nearest_dsep) == {2}

def test_figure_6dot2_2():
    # labels = ["Y", "V1", "V2", "Z"]
    admg = {"-->": [(1, 2), (1, 3), (2, 3)], "<->": [(0, 2)]} 
    x = 0
    y = 3

    nearest_dsep = find_nearest_separator(admg, [x], [y], [], list(range(4)))
    assert nearest_dsep is not None
    assert set(nearest_dsep) == {1, 2}

def test_figure_6dot2_3():
    # labels = ["Y", "V1", "V2", "V3", "V4", "V5", "Z"]
    admg = {"-->": [(1, 2), (2, 5), (2, 6), (3, 2), (4, 3), (4, 0)], "<->": [(0, 1), (0, 5)]} 
    x = 0
    y = 6

    nearest_dsep = find_nearest_separator(admg, [x], [y], [], list(range(6)))
    assert nearest_dsep is not None
    assert set(nearest_dsep) == {1, 4}

File: examples/ciflyr-examples/tests/testthat/testNearestDsep.R
library("here")
source(here("R", "nearestDsep.R"))

# figure from van der Zander (2019) dissertation
test_that("Figure 6.2 G_1", {
	dirEdges <- rbind(c(1, 3), c(2, 1), c(2, 3), c(3, 4), c(4, 5))
	g <- list("-->" = dirEdges)
	x <- 1
	y <- 5

	nearestDsep <- findNearestSeparator(g, c(x), c(y), c(), seq(5))
	expect_equal(sort(nearestDsep), c(2, 3))
})

test_that("Figure 6.2 G_2", {
	dirEdges <- rbind(c(2, 3), c(2, 4), c(3, 4))
	bidirEdges <- rbind(c(1, 3))
	g <- list("-->" = dirEdges, "<->" = bidirEdges)
	x <- 1
	y <- 4

	nearestDsep <- findNearestSeparator(g, c(x), c(y), c(), seq(4))
	expect_equal(sort(nearestDsep), c(2, 3))
})

test_that("Figure 6.2 G_3", {
	dirEdges <- rbind(c(2, 3), c(3, 5), c(3, 7), c(4, 3), c(5, 4), c(5, 1))
	bidirEdges <- rbind(c(1, 2), c(1, 6))
	g <- list("-->" = dirEdges, "<->" = bidirEdges)
	x <- 1
	y <- 7

	nearestDsep <- findNearestSeparator(g, c(x), c(y), c(), seq(6))
	expect_equal(sort(nearestDsep), c(2, 5))
})
